src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py

```diff
@@ -624,13 +624,15 @@ def call(
 
         if inputs["output_hidden_states"]:
             # Add last layer and transpose to library standard shape [bsz, len, hidden_dim]
-            hids.append(core_out)
             hids = tuple(tf.transpose(t, perm=(1, 0, 2)) for t in hids)
+            hids = hids + (core_out,)
+            hids = tf.convert_to_tensor(hids)
```

Comment:
Move the addition of  `core_out` after the `transpose` because `core_out` was already transposed above.